oracle/src/main.rs

```diff
@@ -67,3 +98,51 @@ fn main() -> eyre::Result<()> {
         handle.wait_for_node_exit().await
     })
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::start;
+    use reth_exex_test_utils::test_exex_context;
+
+    #[tokio::test]
+    async fn test_oracle() {
+        reth_tracing::init_test_tracing();
+        let (ctx, _handle) = test_exex_context().await.unwrap();
+        let (oracle, network_1) =
+            start(ctx, 30306, 30307, vec!["btcusdc".to_string(), "ethusdc".to_string()])
+                .await
+                .unwrap();
+        tokio::spawn(oracle);
+
+        // // spawn second
+        // let (ctx, _handle) = test_exex_context().await.unwrap();
+        // let (oracle, network_2) =
+        //     start(ctx, 30308, 30309, vec!["btcusdc".to_string(), "ethusdc".to_string()])
+        //         .await
+        //         .unwrap();
+        // tokio::spawn(oracle);
+
+        // // make them connect
+        // let (peer_1, addr_1) = (network_1.peer_id(), network_1.local_addr());
+        // let (peer_2, addr_2) = (network_2.peer_id(), network_2.local_addr());
+        // network_1.add_peer(*peer_1, addr_1);
+        // network_2.add_peer(*peer_2, addr_2);
+
+        // let mut events_1 = network_1.event_listener();
+
+        // let expected_peer_2 = loop {
```

Comment:
what's missing here, do we need this?

Reply:
> what's missing here, do we need this?

Yup, I am a bit struggling with passing each test `ExExContext` to each oracle instance to finalise this e2e as when you pass it down it gets dropped and the NetworkManager with it.

I think it needs to be restructured in order to be passed around and kept alive as it also has the `ExExNotifications` which requires to be mut.

Do you have any recommendations @mattsse ?

Reply:
where does it get dropped, not really following.

cc @shekhirin testing utils need some work here

Reply:
Yes it's resolved, with https://github.com/paradigmxyz/reth/pull/11907

thanks @shekhirin